Add docker and network pod functions to e2e framework

mkimuram opened the pull request:
This PR is to add docker and network pod functions to e2e framework. These functions are originally needed for https://github.com/submariner-io/submariner/pull/1202 , however, as a result of discussion, we came to conclusion that it would be nice to add them to this framework to make room to leverage from other tests.

mkimuram committed:
Add docker and network pod functions to e2e framework

Signed-off-by: Masaki Kimura <masaki.kimura@hitachivantara.com>

sridhargaddam commented on test/e2e/framework/docker.go:
```diff
@@ -0,0 +1,73 @@
+	var stderr bytes.Buffer
+	// get stderr of `docker log {d.rame}` command
+	cmd := exec.Command("docker", "logs", d.Name)
+	cmd.Stderr = &stderr
```
Why only stderr and not stdout?

mangelajo replied:
:+1:

mangelajo reviewed (commented):
Looks good, only the sridhar comment.

tpantelis commented on test/e2e/framework/docker.go:
```diff
@@ -0,0 +1,73 @@
+	Name string
+}
+
+func NewFromExistingDocker(name string) *Docker {
```
```suggestion
func New(dockerName string) *Docker {
```

tpantelis commented on test/e2e/framework/network_pods.go:
```diff
@@ -178,6 +188,61 @@ func (np *NetworkPod) CreateService() *v1.Service {
 	return np.framework.CreateTCPService(np.Config.Cluster, np.Pod.Labels[TestAppLabel], np.Config.Port)
 }
 
+// RunCommand run the specified comand in this NetworkPod
```
```suggestion
// RunCommand run the specified command in this NetworkPod
```

tpantelis commented on test/e2e/framework/network_pods.go:
```diff
@@ -178,6 +188,61 @@ func (np *NetworkPod) CreateService() *v1.Service {
+
+// GetLog returns container log from this NetworkPod
+func (np *NetworkPod) GetLog() string {
+	req := KubeClients[np.Config.Cluster].CoreV1().RESTClient().Get().
+		Namespace(np.Pod.Namespace).Name(np.Pod.Name).Resource("pods").SubResource("log")
+
+	req.Param("follow", strconv.FormatBool(false)).
+		Param("container", np.Config.ContainerName).
+		Param("previous", strconv.FormatBool(false)).
+		Param("timestamps", strconv.FormatBool(false))
```
It seems like you could use 

`KubeClients[np.Config.Cluster].CoreV1().Pods(np.Pod.Namespace).GetLogs(np.Pod.Name, &corev1.PodLogOptions{...})`

mkimuram replied:
Good catch. Actually, I also found that I was specifying default values for `.PodLogOptions` (Even `container` isn't necessary, if there is only one). It now become much simple.

tpantelis commented on test/e2e/framework/network_pods.go:
```diff
@@ -404,6 +469,38 @@ func (np *NetworkPod) buildLatencyServerPod() {
+		},
+	}
+
+	pc := KubeClients[np.Config.Cluster].CoreV1().Pods(np.framework.Namespace)
+	var err error
+	np.Pod, err = pc.Create(&customPod)
+	Expect(err).NotTo(HaveOccurred())
+	np.AwaitReady()
```
maybe add some newlines for readability...

```suggestion
	pc := KubeClients[np.Config.Cluster].CoreV1().Pods(np.framework.Namespace)
	
	var err error
	np.Pod, err = pc.Create(&customPod)
	Expect(err).NotTo(HaveOccurred())
	
	np.AwaitReady()
```

mkimuram commented:
@mangelajo @sridhargaddam @tpantelis 

Thank you for your reviews. I've fixed and squashed the commit.
I tested locally and confirmed that it works well with the submariner side's PR with "Fix for shipyard framework changes" commit.

PTAL

mangelajo commented:
Great work thanks @mkimuram

tpantelis committed:
Merge branch 'devel' into exte2efunc

sridhargaddam reviewed (approved):
Great work @mkimuram
